qa/1286

```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+# PCP QA Test No. 1286
+# Exercise the BCC PMDA netproc hits module - install, remove and values.
+#
+# Copyright (c) 2020 Red Hat.
+#
+
+seq=`basename $0`
+echo "QA output created by $seq"
+
+. ./common.bcc
+
+_pmdabcc_check
+_pmdabcc_require_kernel_version 4 6
+which curl >/dev/null 2>&1 || _notrun "No curl binary installed"
+which dig >/dev/null 2>&1 || _notrun "No dig binary installed"
+
+status=1       # failure is the default!
+signal=$PCP_BINADM_DIR/pmsignal
+$sudo rm -rf $tmp.* $seq.full
+
+target_http=http://1.1.1.1
+target_dns=1.1.1.1
+
+_value_filter()
+{
+    awk '/"'"$1"'"/ && /value [1-9][0-9]*/ {print "OK"; exit}'
+}
+
+_prepare_pmda bcc
+trap "_pmdabcc_cleanup; exit \$status" 0 1 2 3 15
+_stop_auto_restart pmcd
+
+# real QA test starts here
+cat <<EOF | _pmdabcc_install
+[pmda]
+modules = netproc
+prefix = bcc.
+[netproc]
+module = netproc
+cluster = 40
+remove_stopped_processes = False
+EOF
+_pmdabcc_wait_for_metric
+
+# Generate system activity for the BCC netproc module
+# TCP
+curl -s ${target_http} > /dev/null &
+pid_curl=$!
+echo "curl pid: ${pid_curl}" >> "$seq.full"
+wait ${pid_curl}
+
+# UDP
+# dig performs the DNS request from the first forked process
+pid_dig=$(strace -f -e 'trace=!all' dig @$target_dns -x $target_dns 2>&1 | awk '/Process .* attached/ { print $3; exit }')
```

Comment:
This is a bashism but this is a /bin/sh script.  Also, we need a notrnun check on strace here I guess.

Reply:
hm, I'm using shellcheck which gives you all kinds of linting warnings on shellscripts [1]
afaics this syntax is already a POSIX standard and the old syntax is not recommended anymore ("Because of these inconsistent behaviors, the backquoted variety of command substitution is not recommended for new applications that nest command substitutions or attempt to embed complex scripts.") [2] [3]

Are there shells which can't parse this new syntax? I'm fine changing it to backticks to be in sync with the other shellscripts, but this feels like a (granted, a very tiny) step backwards - shouldn't we go the other direction and embrace the new syntax everywhere instead?

[1] https://www.shellcheck.net/
[2] https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_xcu_chap02.html#tag_23_02_06_03
[3] https://github.com/koalaman/shellcheck/wiki/SC2006

Reply:
I tried a couple and csh doesn't seem to accept it - I dunno if anyone uses that for /bin/sh though.  It mainly stood out cos we just don't use that syntax anywhere else and backticks are pretty well understood.

Reply:
okay, I've updated it.